.github/workflows/release.yml

```
@@ -1,64 +1,150 @@
     steps:
-    - uses: swift-actions/setup-swift@v1
+    - uses: swift-actions/setup-swift@v2
       with:
-        swift-version: "5.8"
-    - uses: actions/checkout@v3
+        swift-version: ${{ env.SWIFT_VERSION }}
+    - uses: actions/checkout@v4
     - name: Create the binary
       run: ./install-script.sh -s . -t mockolo -d . -o mockolo.${{ matrix.destination.name }}.tar.gz
     - name: Upload the binary
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: mockolo.${{ matrix.destination.name }}
         path: mockolo.${{ matrix.destination.name }}.tar.gz
+        name: mockolo.${{ matrix.destination.name }}
+
+  build-with-qemu:
+    name: Build for ${{ matrix.destination.name }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        destination:
+          - { name: "ubuntu-aarch64" }
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - uses: actions/checkout@v4
+      - name: Create the binary
+        run: |
+          docker run --platform linux/arm64 --rm -v ${{ github.workspace }}:/work -w /work swift:${{ env.SWIFT_VERSION }} \
+            ./install-script.sh -s . -t mockolo -d /work -o mockolo.${{ matrix.destination.name }}.tar.gz
+      - name: Upload the binary
+        uses: actions/upload-artifact@v4
+        with:
+          path: mockolo.${{ matrix.destination.name }}.tar.gz
+          name: mockolo.${{ matrix.destination.name }}
 
   check-portability:
-    needs: build-release
-    name: TestRun on ${{ matrix.destination.os }}
+    needs: build
+    name: TestRun on ${{ matrix.destination.os }} for ${{ matrix.destination.name }}
     runs-on: ${{ matrix.destination.os }}
     strategy:
       matrix:
         destination:
-          - { name: "ubuntu", os: ubuntu-22.04 }
-          - { name: "ubuntu", os: ubuntu-20.04 }
-          - { name: "macos", os: macos-12 }
-          - { name: "macos", os: macos-11 }
+          - { name: "ubuntu-x86_64", os: ubuntu-22.04 }
+          - { name: "ubuntu-x86_64", os: ubuntu-20.04 }
+          - { name: "macos-universal", os: macos-14 }
+          - { name: "macos-universal", os: macos-13 }
+          - { name: "macos-universal", os: macos-12 }
     steps:
-    - uses: actions/download-artifact@v3
+    - uses: actions/download-artifact@v4
       with:
         name: mockolo.${{ matrix.destination.name }}
     - name: Unpack the binary
       run: tar -xvf mockolo.${{ matrix.destination.name }}.tar.gz
     - name: Run the binary
       run: ./mockolo --version
 
-  deploy-binary:
-    needs: check-portability
+  check-portability-with-qemu:
+    needs: build-with-qemu
+    name: TestRun on ${{ matrix.destination.os }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        destination:
+          - { name: "ubuntu-aarch64", tag: "ubuntu:latest" }
```

[question]

```suggestion
          - { name: "ubuntu-aarch64", tag: "ubuntu:20.04" }
          - { name: "ubuntu-aarch64", tag: "ubuntu:22.04" }
```

Don't we have to check portability on ubuntu 20 as well?